### PR TITLE
add == for vectors

### LINKED ISSUE #246: Missing method for == on sparse vectors
The following tries to iterate all elements of the sparse vector and takes forever:
```
julia> using SparseArrays

julia> a = SparseVector(div(typemax(Int), 2), [1], [1])
4611686018427387903-element SparseVector{Int64, Int64} with 1 stored entry:
  [1]  =  1

julia> a == a
```

### PR DESCRIPTION
**Pull request by SobhanMP**
merge on green.
Closes #246

### CODE REVIEW

**Comment by dkarrasch** [comment]
This reminds of our broadcast code. Couldn't we replace the code by `all(A .== B)` after the size check?

**Comment by SobhanMP** [comment]
```julia
julia> using SparseArrays

julia> a = sprandn(100000, 0.9);

julia> b = sprandn(100000, 0.9);

julia> @time a .== b;
  0.002611 seconds (7 allocations: 879.344 KiB)
```
i agree that they are similar but that would make it allocating. i don't think we want that.

**Comment by dkarrasch** [comment]
What I had in mind is something lazy like `all(Broadcast.broadcasted(==, Y, Y))`, which is allocation-free, but doesn't seem to use any sparse technology. It seems to just iterate via slow `getindex`.

**Comment by dkarrasch** [comment]
Do we have tests for this? Including things like equal vectors where one has a stored zero?

**Comment by SobhanMP** [comment]
i added a test with explicit zero

**Comment by SobhanMP** [comment]
if you have something specific in mind, do share

File: test/sparsematrix_ops.jl [hunk 1]
```diff
@@ -4,7 +4,7 @@ module SparseTests
 
 using Test
 using SparseArrays
-using SparseArrays: getcolptr, nonzeroinds, _show_with_braille_patterns, _isnotzero
+using SparseArrays: getcolptr, nonzeros, nonzeroinds, _show_with_braille_patterns, _isnotzero
```
Comment (dkarrasch) [inline test/sparsematrix_ops.jl:7]: Seems like `nonzeros` is already exported.

```suggestion
using SparseArrays: getcolptr, nonzeroinds, _show_with_braille_patterns, _isnotzero
```

**Comment by ViralBShah** [comment]
Do we really need all the `@inbounds`? Does it have a noticeable impact here?

**Comment by SobhanMP** [comment]
@ViralBShah on really large examples, yeah

```julia
julia> @benchmark eq1($A, $B)
BenchmarkTools.Trial: 29 samples with 1 evaluation.
 Range (min … max):  168.045 ms … 186.107 ms  ┊ GC (min … max): 0.00% … 0.00%
 Time  (median):     173.698 ms               ┊ GC (median):    0.00%
 Time  (mean ± σ):   174.602 ms ±   4.069 ms  ┊ GC (mean ± σ):  0.00% ± 0.00%

  ▁     ▁▁▁▁█  ██ ▁ █▁▁ ▁▁▁█  ▁▁   ▁ ▁   ▁  ▁      ▁          ▁  
  █▁▁▁▁▁█████▁▁██▁█▁███▁████▁▁██▁▁▁█▁█▁▁▁█▁▁█▁▁▁▁▁▁█▁▁▁▁▁▁▁▁▁▁█ ▁
  168 ms           Histogram: frequency by time          186 ms <

 Memory estimate: 0 bytes, allocs estimate: 0.

julia> @benchmark eq2($A, $B)
BenchmarkTools.Trial: 23 samples with 1 evaluation.
 Range (min … max):  189.635 ms … 370.411 ms  ┊ GC (min … max): 0.00% … 0.00%
 Time  (median):     200.284 ms               ┊ GC (median):    0.00%
 Time  (mean ± σ):   221.056 ms ±  49.187 ms  ┊ GC (mean ± σ):  0.00% ± 0.00%

   █▅                                                            
  ▅███▁██▅▁▁▁▁▅▁▁▁▁▅▅▅▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▅▁▅ ▁
  190 ms           Histogram: frequency by time          370 ms <
```

**Comment by SobhanMP** [comment]
i could replace the last two loops with 
```julia
return all(iszero, view(nonzeros(A), i:nnz(A))) && all(iszero, view(nonzeros(B), j:nnz(B)))
```

**Review by ViralBShah — APPROVED** [review]

**Comment by SobhanMP** [comment]
@ViralBShah so do i replace or no?

**Comment by ViralBShah** [comment]
I think either way is fine.

**Pull request merged**